report/main.py

```diff
@@ -0,0 +1,476 @@
+    """
+    Returns redshift database transaction table as a DataFrame Object
+
+    Args:
+        conn_postgres (connection): A connection to a Postgres database
+
+    Returns:
+        DataFrame:  A pandas DataFrame containing all relevant release data
+    """
+    query = f""
+    df_releases = pd.read_sql_query(query, conn_postgres)
+
+    return df_releases
+
+
+def convert_html_to_pdf(source_html, output_filename):
+    """
+    """
+    # open output file for writing (truncated binary)
+    result_file = open(output_filename, "w+b")
+
+    # convert HTML to PDF
+    pisa_status = pisa.CreatePDF(
+        source_html,                # the HTML to convert
+        dest=result_file)           # file handle to recieve result
+
+    # close output file
+    result_file.close()                 # close output file
+
+    return pisa_status.err
+
+
+def get_new_releases(df_releases: DataFrame) -> int:
+    """
+    Return the number of new releases for the previous day
+
+    Args: 
+        df_releases (DataFrame): A pandas DataFrame containing all relevant game data
+
+    Returns:
+        int: An integer relating to the number of new games released
+    """
+    date = (datetime.now() - timedelta(days=1)).strftime("%Y-%m-%d")
+
+    return df_releases[df_releases["release_date"] == date].reset_index().shape[0]
+
+
+def get_top_rated_release(df_releases: DataFrame) -> str:
+    """
+    Return the name of new release with the highest overall sentiment score for the previous day
+
+    Args: 
+        df_releases (DataFrame): A pandas DataFrame containing all relevant game data
+
+    Returns:
+        str: A string relating to the title of the highest rated new game released
+    """
+    date = (datetime.now() - timedelta(days=1)).strftime("%Y-%m-%d")
+    df_releases = df_releases[df_releases["release_date"] == date]
+    df_ratings = df_releases.groupby("title")["sentiment"].mean().reset_index()
+
+    return df_ratings.head(1)["title"][0]
+
+
+def trending_game_information(df_releases: DataFrame, index: int) -> str:
+    """
+    Return information of new releases with the highest overall sentiment score for the previous day
+
+    Args: 
+        df_releases (DataFrame): A pandas DataFrame containing all relevant game data
+
+        index (int): A int associated with a number index within the trending game list
+
+    Returns:
+        str: A string relating to the information of selected trending new game released in html format
+    """
+    date = (datetime.now() - timedelta(days=1)).strftime("%Y-%m-%d")
+    df_releases = df_releases[df_releases["release_date"] == date]
+    df_trending_game = df_releases.groupby(
+        "title")["sentiment"].mean().reset_index().iloc[index]
+    release_name = df_trending_game["title"]
+    sentiment = round(df_trending_game["sentiment"], 1)
+    release_df = df_releases[df_releases["title"]
+                             == f"{release_name}"].reset_index().head(1)
```

Comment:
Curiosity Q: Oo what is this single equal sign then double equal sign!

Reply:
@gawsalya , in here he sets release_df to be a part of df_releases where the title is specifically the release_name variable